Codecademy/gamut
# fix(FormGroupLabel): enable screenreaders to read a label's "optional" text

### Overview


Currently, Gamut's forms' labels include either `*` to signify required, or `(optional)` to tell learners what fields are necessary. 
Both are set to `aria-hidden` but a test.io filing requests that the `(optional)` be read aloud.

This PR changes the `(optional)` text to be read along with the label text and keeps the `aria-hidden` on the `*`. 


### PR Checklist

- [x] Related to JIRA ticket: [GM-910]

#### Testing Instructions



1. Go to the Preview link
2. Go to FormGroupLabel: ?path=/docs/atoms-formelements-formgrouplabel--docs&globals=viewport:responsive
3. Turn on VO and check that the `optional` text is being read aloud
4. NOTE: you might need "control" + "option" + arrow key to access the text since it's a non-interactive item
5. Optionally, but appreciated (6 + 7):
6. Repeat for an example in ConnectedForm: ?path=/docs/organisms-connectedform-connectedform--docs 
7. Repeat for  an example in GridForm: ?path=/docs/organisms-gridform--docs 
8. ...
9. Profit! 

—-
for mono:
1. Open the portal-app preview and log in
2. Go to the Profile page, click "Edit Profile" under your profile picture + name (e.g. slug should be `profiles/<username>/edit`
3. Turn on VoiceOver
4. Navigate over the fields that have "(optional)" text in them, "optional" should now be read aloud and along with the label text
5. Navigate over the required fields' labels, the "*" should still not be annouced. 
6. ...
7. Profit!

—- 
for monolith:
1. Open the preview and log in
2. Go to the "Account + Billing" page by clicking on your avatar (e.g. slug should be `/account`)
3. Click "Change" for either email or password
4. Turn on VoiceOver
5. Navigate over the required fields' labels, the "*" should still not be annouced. 
6. Go to the "Goal settings" tab
7. Navigate over the fields that have "(optional)" text in them, "optional" should now be read aloud and along with the label text
8. ...
9. Profit!

—-

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/38954)  | [Monolith Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-8845) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/8845)  | [Portal Env](https://pr-38954-monolith.dev-eks.codecademy.com/)   |





[GM-910]: https://skillsoftdev.atlassian.net/browse/GM-910?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
